### PR TITLE
Fix #125: Test server: endpoint for configuration of application

### LINKED ISSUE #125: Test server: endpoint for configuration of application
Currently, the application on test server needs to be configured manually. This means that SQL inserts are required before first test run. We can automate the configuration by preparing a `POST /application/config` endpoint which will enable configuration of the test server using REST API.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/controller/ApplicationController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public ApplicationController(ApplicationService applicationService) {
+        this.applicationService = applicationService;
+    }
+
+    /**
+     * Configure an application.
+     * @param request Configure an application request.
+     * @return Configure an application response.
+     */
+    @RequestMapping(value = "config", method = RequestMethod.POST)
```
Comment (banterCZ) [inline powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/controller/ApplicationController.java:56]:
```suggestion
    @PostMapping("config")
```

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/model/request/ConfigureApplicationRequest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Data
+public class ConfigureApplicationRequest {
+
+    private String applicationId;
+    private String applicationName;
+    private String applicationKey;
+    private String applicationSecret;
```
Comment (banterCZ) [inline powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/model/request/ConfigureApplicationRequest.java:34]: What about `applicationSecret` included in `toString`?

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ApplicationService.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        final String applicationKey = request.getApplicationKey();
+        final String applicationSecret = request.getApplicationSecret();
+        final String masterPublicKey = request.getMasterPublicKey();
+
+        TestConfigEntity appConfig;
+        try {
+            appConfig = getTestAppConfig(applicationId);
+        } catch (AppConfigNotFoundException ex ) {
+            appConfig = new TestConfigEntity();
+            appConfig.setApplicationId(request.getApplicationId());
```
Comment (banterCZ) [inline powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ApplicationService.java:66]:
```suggestion
            appConfig.setApplicationId(applicationId);
```

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ApplicationService.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    @Transactional
+    public Response configureApplication(ConfigureApplicationRequest request) {
+        final String applicationId = request.getApplicationId();
+        final String applicationName = request.getApplicationName();
+        final String applicationKey = request.getApplicationKey();
+        final String applicationSecret = request.getApplicationSecret();
+        final String masterPublicKey = request.getMasterPublicKey();
+
+        TestConfigEntity appConfig;
```
Comment (banterCZ) [inline powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ApplicationService.java:61]: What about to extract the try/catch block to a separate method `getOrCreateTestAppConfig`, make variable final, and log whether entity was created or updated?

**Review by banterCZ — APPROVED** [review]
Great! I have just some little suggestions.

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/controller/ApplicationController.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/controller/ApplicationController.java:56]: Sure, I can change it.

**Pull request merged**